### PR TITLE
Creating Noobaa SCC when installing using cli

### PR DESCRIPTION
**Pull request by jackyalbo**
Will create Noobaa SCC if not found and SCC CRD exist
Added option to uninstall SCC by running: ```noobaa uninstall --cleanup ```

Signed-off-by: jackyalbo <jalbo@redhat.com>

### CODE REVIEW

File: deploy/security_context_constraints.yaml [hunk 1]
```diff
@@ -0,0 +1,36 @@
+apiVersion: security.openshift.io/v1
```
Comment (guymguym) [inline deploy/security_context_constraints.yaml:1]: I would rename the file to scc.yaml - I wouldn't expect to find it with this full name and might miss it

File: pkg/install/install.go [hunk 2]
```diff
@@ -34,6 +34,7 @@ func CmdUninstall() *cobra.Command {
 	}
 	cmd.Flags().Bool("cleanup", false, "Enable deletion of Namespace and CRD's")
 	cmd.Flags().Bool("cleanup_data", false, "Clean object buckets")
+	cmd.Flags().Bool("cleanscc", false, "Enable deletion of the NooBaa SCC")
```
Comment (guymguym) [inline pkg/install/install.go:37]: why not include in cleanup? it cleans other clusterwide entities

File: pkg/operator/operator.go [hunk 3]
```diff
@@ -210,6 +221,10 @@ func LoadOperatorConf(cmd *cobra.Command) *Conf {
 		c.Deployment.Spec.Template.Spec.ImagePullSecrets =
 			[]corev1.LocalObjectReference{{Name: options.ImagePullSecret}}
 	}
+
+	c.SecurityContextConstraints.Name = c.SA.Name
+	c.SecurityContextConstraints.Namespace = options.Namespace
```
Comment (guymguym) [inline pkg/operator/operator.go:226]: namespace? for SCC? seems irrelevant and might cause errors.

File: pkg/operator/operator.go [hunk 4]
```diff
@@ -210,6 +221,10 @@ func LoadOperatorConf(cmd *cobra.Command) *Conf {
 		c.Deployment.Spec.Template.Spec.ImagePullSecrets =
 			[]corev1.LocalObjectReference{{Name: options.ImagePullSecret}}
 	}
+
+	c.SecurityContextConstraints.Name = c.SA.Name
```
Comment (guymguym) [inline pkg/operator/operator.go:225]: why do we need to set the name dynamically and not in the yaml?

File: pkg/operator/operator.go [hunk 5]
```diff
@@ -94,6 +96,7 @@ func RunInstall(cmd *cobra.Command, args []string) {
 	util.KubeCreateSkipExisting(c.RoleBinding)
 	util.KubeCreateSkipExisting(c.ClusterRole)
 	util.KubeCreateSkipExisting(c.ClusterRoleBinding)
+	util.KubeCreateSkipExisting(c.SecurityContextConstraints)
```
Comment (guymguym) [inline pkg/operator/operator.go:99]: Actually is this where you wanted to add it at all? This only runs when calling `noobaa install` and is not running during reconcile at all. This is fine if we just want to support CLI install of the SCC, but this means that we still do not handle it instead of OCS operator.

Comment (guymguym) [reply pkg/operator/operator.go:99]: Also this will fail on non openshift clusters, where the SCC CRD is not available.
See how we create other openshift CR's like Routes.

File: pkg/install/install.go (hunk 2 above)
Comment (jackyalbo) [reply pkg/install/install.go:37]: IDK just for a shared cluster like gogo - I accidentally deleted the SCC there

File: pkg/operator/operator.go (hunk 5 above)
Comment (jackyalbo) [reply pkg/operator/operator.go:99]: We want to install during cli install and that's it for now - reconciles will still be only in OCS operator
Tried in minikube and it worked fine. Just warned that the SCC is misssing: 
```INFO[0002] ❌ CRD Missing: SecurityContextConstraints "noobaa"```

File: pkg/operator/operator.go (hunk 4 above)
Comment (jackyalbo) [reply pkg/operator/operator.go:225]: right, I can remove both

File: pkg/operator/operator.go (hunk 3 above)
Comment (jackyalbo) [reply pkg/operator/operator.go]: moved to cleanup

**Review by dannyzaken — APPROVED** [review]

File: pkg/operator/operator.go (hunk 5 above)
Comment (guymguym) [reply pkg/operator/operator.go:99]: this looks like an error to me. that's why we used the ReconcileObjectOptional -
https://github.com/noobaa/noobaa-operator/blob/47283a1777a0a42567de44a608771de7981ca0aa/pkg/system/reconciler.go#L522
and the CheckOptional -
https://github.com/noobaa/noobaa-operator/blob/47283a1777a0a42567de44a608771de7981ca0aa/pkg/util/util.go#L366
to avoid these things looking like errors to the user.

Comment (jackyalbo) [reply pkg/operator/operator.go]: Added KubeCreateOptional
